builtin/providers/nsone/resource_record.go

```diff
@@ -0,0 +1,854 @@
+					"value": strings.Join(ipPrefixes, ","),
+				})
+			}
+		default:
+			panic(fmt.Sprintf("meta.IPPrefixes in unexpected type: %#v", meta.IPPrefixes))
+		}
+		switch meta.ASN.(type) {
+		case nil:
+			break
+		case []string:
+			asn := meta.ASN.([]string)
+			if len(asn) != 0 {
+				sort.Strings(asn)
+				metas.Add(map[string]interface{}{
+					"field": "asn",
+					"value": strings.Join(asn, ","),
+				})
+			}
+		default:
+			panic(fmt.Sprintf("meta.ASN in unexpected type: %#v", meta.ASN))
+		}
+		switch meta.Priority.(type) {
+		case nil:
+			break
+		case int:
+			priority := meta.Priority.(int)
+			if priority != 0 {
+				metas.Add(map[string]interface{}{
+					"field": "priority",
+					"value": strconv.Itoa(priority),
+				})
+			}
+		default:
+			panic(fmt.Sprintf("meta.Priority in unexpected type: %#v", meta.Priority))
+		}
+		switch meta.Weight.(type) {
+		case nil:
+			break
+		case float64:
+			weight := meta.Weight.(float64)
+			if weight != 0 {
+				metas.Add(map[string]interface{}{
+					"field": "weight",
+					"value": strconv.Itoa(int(weight)),
+				})
+			}
+		default:
+			panic(fmt.Sprintf("meta.Weight in unexpected type: %#v", meta.Weight))
+		}
+		switch meta.LowWatermark.(type) {
+		case nil:
+			break
+		case int:
+			lowWatermark := meta.LowWatermark.(int)
+			if lowWatermark != 0 {
+				metas.Add(map[string]interface{}{
+					"field": "low_watermark",
+					"value": strconv.Itoa(lowWatermark),
+				})
+			}
+		default:
+			panic(fmt.Sprintf("meta.LowWatermark in unexpected type: %#v", meta.LowWatermark))
+		}
+		switch meta.HighWatermark.(type) {
+		case nil:
+			break
+		case int:
+			highWatermark := meta.HighWatermark.(int)
+			if highWatermark != 0 {
+				metas.Add(map[string]interface{}{
+					"field": "high_watermark",
+					"value": strconv.Itoa(highWatermark),
+				})
+			}
+		default:
+			panic(fmt.Sprintf("meta.HighWatermark in unexpected type: %#v", meta.HighWatermark))
+		}
+		m["meta"] = metas
+	}
+	return m
+}
+
+func btoi(b bool) int {
+	if b {
+		return 1
+	}
+	return 0
+}
+
+func resourceDataToRecord(r *dns.Record, d *schema.ResourceData) error {
+	r.ID = d.Id()
+	if answers := d.Get("answers").(*schema.Set); answers.Len() > 0 {
+		al := make([]*dns.Answer, answers.Len())
+		for i, answerRaw := range answers.List() {
+			answer := answerRaw.(map[string]interface{})
+			var a *dns.Answer
+			v := answer["answer"].(string)
+			switch d.Get("type") {
+			case "TXT":
+				a = dns.NewTXTAnswer(v)
+			default:
+				a = dns.NewAnswer(strings.Split(v, " "))
+			}
+			if v, ok := answer["region"]; ok {
+				a.RegionName = v.(string)
+			}
+			if metas := answer["meta"].(*schema.Set); metas.Len() > 0 {
+				for _, metaRaw := range metas.List() {
+					meta := metaRaw.(map[string]interface{})
+					key := meta["field"].(string)
+					if value, ok := meta["feed"]; ok && value.(string) != "" {
+						switch key {
+						case "up": // bool
+							a.Meta.Up = data.NewFeed(value.(string), data.Config{})
```

Comment:
These should be:
```
a.Meta.Up = data.FeedPtr{value.(string)}
```

The `data.NewFeed` constructs a new datafeed, whereas we want a 'feed pointer'